File: 프로그래머스/unrated/181925. 수 조작하기 2/수 조작하기 2.kt
class Solution {
    fun solution(numLog: IntArray): String {
    val result = StringBuilder()
    var currentNum = numLog[0]

    for (i in 1 until numLog.size) {
        val diff = numLog[i] - numLog[i - 1]

        when (diff) {
            1 -> result.append('w')
            -1 -> result.append('s')
            10 -> result.append('d')
            -10 -> result.append('a')
        }

        currentNum = numLog[i]
    }

    return result.toString()
}
}

File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.js
function solution(numLog) {
    return numLog
        .map((val, i ,arr) => {
        if(i===0) return '';
        
        const diff = val - arr[i-1];
        if(diff === 1 ) return 'w';
        if(diff === -1) return 's';
        if(diff === 10) return 'd';
        if(diff === -10) return 'a';
    })
    .join('')
}